xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -92,18 +92,33 @@ impl DbConnection<'_> {
         Ok(())
     }
 
+    /// Updates the 'last time checked' for installation lists.
+    /// @returns Result<(previously_checked_ts, latest_checked_ts)>
     pub fn update_installation_list_time_checked(
         &self,
         group_id: Vec<u8>,
-    ) -> Result<i64, StorageError> {
-        let now = crate::utils::time::now_ns();
-        self.raw_query(|conn| {
+    ) -> Result<(i64, i64), StorageError> {
+        let (last_ts, latest_ts) = self.raw_query(|conn| {
+            let maybe_get_last_ts = dsl::groups
+                .find(&group_id)
+                .select(dsl::installation_list_last_checked)
+                .execute(conn)
+                .optional()?;
+
+            let now = crate::utils::time::now_ns();
+            let last_ts = match maybe_get_last_ts {
+                Some(ts) => ts as i64,
+                None => now,
```

Comment:
I think `last_ts` should just be 0 if it was not previously set, that way we are guaranteed to check on the first time.

With this in mind, I think the DB schema can have `installation_list_last_checked` be a non-nullable value with a default of 0, so that we don't have to deal with an `Option` here. Also, I think we should return an error if the `group_id` doesn't exist in the DB, so that bugs can be caught here.

Reply:
Sure, I guess we don't need to care about an un-initialized state.
I made it a `NOT NULL` and initializes the value to `0`.  And fixed for the `NotFound` Err case.